### PR TITLE
Set URL of target project when branching from SCM/CI

### LINKED ISSUE #11552: Add link back to git in the PR projects
I would like to have a link to https://github.com/openSUSE/open-build-service/pull/11489 in https://build.opensuse.org/project/show/home:bs-team:OBS:Server:Unstable:PR-11489

The project description says atm "This project was created for package obs-server via attribute OBS:Maintained" and I think that's wrong and having a proper link could help.

BTW: it would be great if those old projects could be removed, they cause a lot of build churn (and in this very case emails for build failures in my inbox)

### PR DESCRIPTION
**Pull request by eduardoj**
Fixes #11552.

### CODE REVIEW

File: src/api/app/models/workflow/step/branch_package_step.rb [hunk 1]
```diff
@@ -24,6 +25,10 @@ def call
 
   private
 
+  def modify_target_project_url
+    target_project.update(url: workflow_run.event_source_url)
```
Comment (hennevogel) [inline src/api/app/models/workflow/step/branch_package_step.rb:29]: Unlike `Package`, `Project` has to be explicitly written to the backend...

Comment (eduardoj) [reply src/api/app/models/workflow/step/branch_package_step.rb]: True. Please take a look again.

File: src/api/app/models/workflow/step/branch_package_step.rb [hunk 2]
```diff
@@ -48,6 +48,10 @@ def target_project
     Project.find_by(name: target_project_name)
   end
 
+  def target_project_url
```
Comment (hennevogel) [inline src/api/app/models/workflow/step/branch_package_step.rb:51]: This method makes not much sense and just hides what this really is below...

File: src/api/app/models/branch_package.rb [hunk 3]
```diff
@@ -263,13 +263,13 @@ def create_branch_project
 
       title = "Branch project for package #{params[:package]}"
       description = "This project was created for package #{params[:package]} via attribute #{@attribute}"
-      base_project_url = Project.find_by_name(params[:project]).try(:url)
+      url = params[:target_project_url] || Project.find_by_name(params[:project]).try(:url)
```
[the remark this answers is not in the document]
Comment (eduardoj) [reply src/api/app/models/branch_package.rb:266]: Yes. :sweat_smile: 

What do you really mean?

[the remark this answers is not in the document]
Comment (hennevogel) [reply src/api/app/models/branch_package.rb:266]: I *really* mean nice :)

**Review by hennevogel — APPROVED** [review]

File: src/api/app/models/workflow/step/branch_package_step.rb (hunk 2 above)
Comment (hennevogel) [reply src/api/app/models/workflow/step/branch_package_step.rb]: DRY is not about typing :) The ` WorkflowRun.event_source_url` is perfectly DRY...

**Pull request merged**